test/aws.storage.test.js

```diff
@@ -0,0 +1,140 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire');
+
+const conf = require('../server/config.js');
+conf.notLocalHost = true;
+
+let redisStub = {};
+let exists = sinon.stub();
+let hget = sinon.stub();
+let hmset = sinon.stub();
+let expire = sinon.stub();
+let del = sinon.stub();
+
+redisStub.createClient = function() {
+  return {
+    on: sinon.spy(),
+    exists: exists,
+    hget: hget,
+    hmset: hmset,
+    expire: expire,
+    del: del
+  }
+}
+
+let fsStub = {};
+fsStub.statSync = sinon.stub();
+fsStub.createReadStream = sinon.stub();
+fsStub.createWriteStream = sinon.stub();
+fsStub.unlinkSync = sinon.stub();
+
+let logStub = {};
+logStub.info = sinon.stub();
+logStub.error = sinon.stub();
+
+let s3Stub = {};
+s3Stub.headObject = sinon.stub();
+s3Stub.getObject = sinon.stub();
+s3Stub.upload = sinon.stub();
+s3Stub.deleteObject = sinon.stub();
+
+let awsStub = {
+    S3: function() {
+        return s3Stub;
+    }
+}
+
+const storage = proxyquire('../server/storage', {
+  'redis': redisStub,
+  'fs': fsStub,
+  './log.js': function() { return logStub },
+  'aws-sdk': awsStub
+});
+
+describe('Testing Length using aws', function() {
+  it('Filesize returns properly if id exists', function() {
+    s3Stub.headObject.callsArgWith(1, null, {ContentLength: 1});
+    return storage.length('123')
+                  .then(reply => assert(reply === 1))
```

Comment:
lets prefer `assert.equal` when possible. It'll print better messages when the assertion fails, including the actual and expected values.